src/Nethermind/Nethermind.Network/Rlpx/Handshake/HandshakeService.cs

```diff
@@ -87,10 +88,10 @@ public Packet Auth(PublicKey remoteNodeId, EncryptionHandshake handshake, bool p
                 authMessage.PublicKey = _privateKey.PublicKey;
                 authMessage.Signature = _ecdsa.Sign(handshake.EphemeralPrivateKey, new Keccak(forSigning));
 
-                byte[] authData = _messageSerializationService.Serialize(authMessage);
-                int size = authData.Length + 32 + 16 + 65; // data + MAC + IV + pub
+                IByteBuffer authData = _messageSerializationService.ZeroSerialize(authMessage);
+                int size = authData.ReadableBytes + 32 + 16 + 65; // data + MAC + IV + pub
                 byte[] sizeBytes = size.ToBigEndianByteArray().Slice(2, 2);
-                byte[] packetData = _eciesCipher.Encrypt(remoteNodeId, authData,sizeBytes);
+                byte[] packetData = _eciesCipher.Encrypt(remoteNodeId, authData.ReadAllBytes(),sizeBytes);
```

Comment:
I wonder if we could use `ArrayPool` in such cases, when the array is consumed by encryption and we can immediately return it next line...

Reply:
Yes, its used in a similar way in lot of places, we can change it to ArrayPool, should I do this in this PR of new PR for similar changes?

Reply:
Probably it's an improvement to be considered and made within another issue & PR

Reply:
allocation to die

Reply:
Ok this is tricky as we need exact length here, so a normal array pool will blow the length to 256 when we need only 192, this is then encrypted so it's super important.

Reply:
Better to remove BouncyCastle and use something based on Span